src/components/Header/Form/Form.js

```diff
@@ -0,0 +1,135 @@
+import React, { useState } from 'react';
+import { Button } from '../../Button';
+import { Search } from '../../../Icons';
+import { dataAvailable } from './config';
+
+export function Form({ setResults }) {
+  const [search, setSearch] = useState('');
+
+  function hotelSearch(event) {
+    // const searchValue = event.target.value;
+    setSearch(event.target.value);
+    setResults(
+      dataAvailable.filter(
+        (hotel) =>
+          hotel.name &&
+          search &&
+          hotel.name.toLowerCase().includes(event.target.value.toLowerCase()),
+      ),
+    );
+  }
+
+  return (
+    <form
+      className="header__form"
+      onSubmit={(event) => {
+        event.preventDefault();
+        hotelSearch(event);
+      }}
+    >
+      <Search className="form__destination-search" />
+      <input
+        className="form__destination form__text"
+        type="text"
+        name="hotel"
+        value={search}
+        onChange={(event) => setSearch(event.target.value)}
```

Comment:
your callback working as well like your hotelSearch function, so you better to remove setSearch(event.target.value) inside hotelSearch function